src/lib/signature_lib/test/test_private_key.ml

```diff
@@ -0,0 +1,312 @@
+open Core_kernel
+open Signature_lib
+open Base_quickcheck
+
+let modulus_string =
+  "28948022309329048855892746252171976963363056481941647379679742748393362948097"
+
+let test_of_string_exn_zero () =
+  let key = Private_key.of_string_exn "0" in
+  Alcotest.(check bool)
+    "Private key from string \"0\" should create valid key" true
+    (Snark_params.Tick.Inner_curve.Scalar.equal key
+       Snark_params.Tick.Inner_curve.Scalar.zero )
+
+let test_of_string_exn_one () =
+  let key = Private_key.of_string_exn "1" in
+  Alcotest.(check bool)
+    "Private key from string \"1\" should create valid key" true
+    (Snark_params.Tick.Inner_curve.Scalar.equal key
+       Snark_params.Tick.Inner_curve.Scalar.one )
+
+let test_of_string_exn_small_values () =
+  let gen = Generator.int_inclusive 1 1000000 in
+  let test_values =
+    List.init 5 ~f:(fun _ -> Quickcheck.random_value gen |> Int.to_string)
+  in
+  List.iter test_values ~f:(fun s ->
+      let key = Private_key.of_string_exn s in
```

Comment:
So these 2 functions are just the same function? If so why even bother design them to be 2 functions. instead of alias of each other?

Reply:
> So these 2 functions are just the same function?

Wdym?